### PR TITLE
Kotlin: Fix extraction of `$default` extension functions

### PR DESCRIPTION
**Pull request by tamasvajk**
This PR changes
- the extraction of `$default` methods to mark them as extension functions if their original functions were extensions.
- fixes the order of the dispatch and extension receiver parameters in member extension `$default` methods
- adjusts the `PathSanitizer` library to make use of the fixes.

### CODE REVIEW

**Review by smowton — APPROVED** [review]
Kotlin extractor changes look good (haven't reviewed the path-sanitizer lib)

File: java/ql/lib/semmle/code/java/Member.qll [hunk 1]
```diff
@@ -824,4 +814,19 @@ class ExtensionMethod extends Method {
   KotlinType getExtendedKotlinType() { result = extendedKotlinType }
 
   override string getAPrimaryQlClass() { result = "ExtensionMethod" }
+
+  /**
+   * Gets the index of the parameter that is the extension receiver. This is typically index 0. In case of `$default`
+   * extension methods that are defined as members, the index is 1. Index 0 is the dispatch receiver of the `$default`
+   * method.
+   */
+  int getExtensionParameterIndex() {
```
Comment (atorralba) [inline java/ql/lib/semmle/code/java/Member.qll:823]: Should this be called `getExtensionReceiverParameterIndex`?

File: java/ql/lib/change-notes/2022-12-09-default-extension-methods.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+category: minorAnalysis
+---
+* The extraction of extension methods have been improved when default parameter values are present. The dispatch and extension receiver parameters are extracted in the correct order. The `ExtensionMethod::getExtensionParameterIndex` predicate has been introduced to facilitate getting the correct extension parameter index.
```
Comment (atorralba) [inline java/ql/lib/change-notes/2022-12-09-default-extension-methods.md:4]:
```suggestion
* The extraction of extension methods has been improved when default parameter values are present. The dispatch and extension receiver parameters are extracted in the correct order. The `ExtensionMethod::getExtensionParameterIndex` predicate has been introduced to facilitate getting the correct extension parameter index.
```

Also, should we mention Kotlin somewhere here?

**Review by atorralba — COMMENTED** [review]
Changes in `PathSanitizer.qll` LGTM! 👍 

Added a couple of minor comments.

File: java/ql/lib/change-notes/2022-12-09-default-extension-methods.md (hunk 2 above)
Comment (tamasvajk) [reply java/ql/lib/change-notes/2022-12-09-default-extension-methods.md:4]: Thanks. Let me adjust it.

**Comment by tamasvajk** [comment]
@atorralba I applied your review feedback. (and rebased the PR to fix a merge conflict)

**Review by atorralba — APPROVED** [review]

**Pull request merged**